addons/rating/models/mail_thread.py

```diff
@@ -204,3 +205,17 @@ def _message_update_content(self, message, body, *args, rating_value=None, **kwa
         return super()._message_update_content(
             message, body, *args, rating_value=rating_value, **kwargs
         )
+
+    def _thread_to_store(self, store: Store, fields, *, request_list=None):
+        super()._thread_to_store(store, fields, request_list=request_list)
+        for thread in self:
+            if (
+                request_list
+                and "ratingStats" in request_list
+                and hasattr(thread, "rating_get_stats")
+            ):
```

Comment:
I'm pretty sure the method is necessarily called on threads of the same model, so the condition can be outside of the loop.